ecs-agent/gmsacredclient/credentialsfetcherclient.go

```diff
@@ -62,6 +62,116 @@ type CredentialsFetcherResponse struct {
+	}
+
+	if len(accessKeyId) == 0 || len(secretKey) == 0 || len(sessionToken) == 0 || len(region) == 0 {
+		return CredentialsFetcherArnResponse{}, status.Errorf(codes.InvalidArgument, "accessid, secretkey, sessiontoken or region should not be empty")
+	}
+
+	defer c.conn.Close()
+	client := pb.NewCredentialsFetcherServiceClient(c.conn)
+
+	request := &pb.KerberosArnLeaseRequest{CredspecArns: credentialspecsArns, AccessKeyId: accessKeyId, SecretAccessKey: secretKey, SessionToken: sessionToken, Region: region}
+
+	ctx, cancel := context.WithDeadline(ctx, time.Now().Add(c.timeout))
+	defer cancel()
+
+	response, err := client.AddKerberosArnLease(ctx, request)
+	if err != nil {
+		seelog.Errorf("could not create kerberos tickets: %v", err)
+		return CredentialsFetcherArnResponse{}, err
+	}
+	seelog.Infof("created kerberos tickets and associated with LeaseID: %s", response.GetLeaseId())
+
+	credentialsFetcherResponse := CredentialsFetcherArnResponse{
+		LeaseID:            response.GetLeaseId(),
+		KerberosTicketsMap: make(map[string]string),
+	}
+
+	for _, value := range response.GetKrbTicketResponseMap() {
+		credSpecArn := value.GetCredspecArns()
+		_, ok := credentialsFetcherResponse.KerberosTicketsMap[credSpecArn]
+		if ok {
+			seelog.Infof("credentialspec arn already added to map: %s", credSpecArn)
+		} else {
+			credentialsFetcherResponse.KerberosTicketsMap[credSpecArn] = value.GetCreatedKerberosFilePaths()
+		}
+	}
+
+	return credentialsFetcherResponse, nil
+}
+
+// RenewKerberosArnLease() invokes credentials fetcher daemon running on the host
+// to create kerberos tickets associated with gMSA accounts
+func (c CredentialsFetcherClient) RenewKerberosArnLease(ctx context.Context, credentialspecsArns []string, accessKeyId string, secretKey string, sessionToken string, region string) (string, error) {
+	if len(credentialspecsArns) == 0 {
+		return codes.InvalidArgument.String(), status.Errorf(codes.InvalidArgument, "credentialspecArns should not be empty")
+	}
+
+	if len(accessKeyId) == 0 || len(secretKey) == 0 || len(sessionToken) == 0 || len(region) == 0 {
+		return codes.InvalidArgument.String(), status.Errorf(codes.InvalidArgument, "accessid, secretkey, sessiontoken or region should not be empty")
+	}
+
+	defer c.conn.Close()
+	client := pb.NewCredentialsFetcherServiceClient(c.conn)
+
+	request := &pb.KerberosArnLeaseRequest{CredspecArns: credentialspecsArns, AccessKeyId: accessKeyId, SecretAccessKey: secretKey, SessionToken: sessionToken, Region: region}
+
+	ctx, cancel := context.WithDeadline(ctx, time.Now().Add(c.timeout))
+	defer cancel()
+
+	response, err := client.RenewKerberosArnLease(ctx, request)
+	if err != nil {
+		seelog.Errorf("could not renew kerberos tickets: %v", err)
+		return codes.Internal.String(), err
+	}
+
+	if response.GetStatus() == "failed" {
+		return codes.Internal.String(), status.Errorf(codes.Internal, "renewal of kerberos tickets failed")
+	}
```

Comment:
Does this check not apply to `AddKerberosArnLease` method?

Reply:
add kerberos lease will look at rpc failure or if error message